json-schema-check/action.yaml

```diff
@@ -0,0 +1,15 @@
+name: json-schema-check
+description: ""
+
+runs:
+  using: composite
+  steps:
+    - name: Install JSON Schema packages
+      run: |
+        pip3 install -U check-jsonschema
+      shell: bash
+
+    - name: Check configuration files
+      run: |
+        find -wholename '*/schema/*.schema.json' -execdir bash -c 'check-jsonschema --schemafile "$1" ../config/*.param.yaml' bash '{}' +
```

Comment:
> 1 schema/{pkg_name}.schema.json that acts the template for the configuration files
1 (or more) config/*.param.yaml configuration that need to match the template

It seems these behaviors aren't implemented yet.
Is this pull request still under draft or do you plan to implement them in another pull request?
If it's the former, please mark this pull request as a draft.
If it's the latter, creating an issue would be good.
(Or am I misunderstanding something?)

Reply:
The design for the structure has been published here: https://github.com/orgs/autowarefoundation/discussions/3371
It has been implemented as an example for [lidar_apollo_segmentation_tvm_nodes](https://github.com/autowarefoundation/autoware.universe/blob/435cdf909bafba40832472a5337cf72127a50ba3/perception/lidar_apollo_segmentation_tvm_nodes/schema/lidar_apollo_segmentation_tvm_nodes.schema.json). A second one was added recently with [autoware_auto_msgs_adapter](https://github.com/autowarefoundation/autoware.universe/blob/435cdf909bafba40832472a5337cf72127a50ba3/system/autoware_auto_msgs_adapter/schema/autoware_auto_msgs_adapter.schema.json). The rest is still to be done.
The idea is that in order for people to add those files, it is valuable to have the format checked in CI before they start to create patches. Otherwise we risk having incorrect files in the repositories and it will take time later to fix.

Reply:
Ah, I misunderstood one thing. The correspondence between `param.yaml` and `schema.json` was checked in the command.

@ambroise-arm So do you want to merge this as it is?
I mean, it seems that this action succeeds if a package doesn't have any `schema.json`. Is that okay for you?

Reply:
I will update the spelling with ROS 2, otherwise yes.
Should I update .github/workflows/test-composite-actions.yaml and the main README to add this? Or anything else that I missed?

Reply:
> Should I update .github/workflows/test-composite-actions.yaml

It's not mandatory, but it's preferable if you can test this action.

> the main README to add this?

Ah, yes. It would be nice.

> Or anything else that I missed?

No, looks good! Thank you for your work.

I'll make a new release after merging it.

Reply:
I've fixed them, so I'll merge this.